internal/gatewayapi/runner/runner.go

```diff
@@ -182,6 +205,153 @@ func (r *Runner) deleteAllIRKeys() {
 	}
 }
 
+type StatusesToDelete struct {
+	GatewayStatusKeys          map[types.NamespacedName]bool
+	HTTPRouteStatusKeys        map[types.NamespacedName]bool
+	GRPCRouteStatusKeys        map[types.NamespacedName]bool
+	TLSRouteStatusKeys         map[types.NamespacedName]bool
+	TCPRouteStatusKeys         map[types.NamespacedName]bool
+	UDPRouteStatusKeys         map[types.NamespacedName]bool
+	BackendTLSPolicyStatusKeys map[types.NamespacedName]bool
+
+	ClientTrafficPolicyStatusKeys  map[types.NamespacedName]bool
+	BackendTrafficPolicyStatusKeys map[types.NamespacedName]bool
+	SecurityPolicyStatusKeys       map[types.NamespacedName]bool
+}
```

Comment:
maybe we should use set[T] instead of map[T]bool, but we can do that in the future.